audformat/core/utils.py

```diff
@@ -1074,3 +1074,24 @@ def union(
     index = index.drop_duplicates()
 
     return index
+
+
+def iter_index_by_file(
+        index: pd.Index
+) -> (str, typing.Iterator[pd.Index]):
+    r"""Iterate over index by file.
+
+    In each iteration return sub-index for one file.
+
+    Args:
+        index: index object conform to
+            :ref:`table specifications <data-tables:Tables>`
+
+    Returns:
+        iterator in form of (file, sub_index)
+    """
+    index = to_segmented_index(index, allow_nat=False)
```

Comment:
First of all, I don't think we should force `allow_nat=False` here by default. It can be a costly. But I'm actually not sure if we should convert to a segmented index at all here, bust instead add support for `filewise` indices. It does not make so much sense at the moment, since we currently don't allow duplicates, so it will return exactly one file in each iteration, but maybe we will allow duplicates in the future again. Generally, I'd like to leave it to the user to convert to a segmented index if needed.